### DISCUSSION TITLE
Adding header at factory creation?

### DISCUSSION THREAD

**Discussion by dannythunder** [canton7/RestEase#241] [Q&A] [answered]
How do I add an header when creating the client?

I wan't to add a default dynamic header (apikey), but I can't find out how in the Readme.

`services.AddRestEaseClient<IMyHttpClient>(baseUrl);`

I would like to chain `.AddDefaultHeader("key", "value"); 
or something like this:

```
services.AddRestEaseClient<IMyHttpClient>()
    .Configure(client =>
        client.BaseUrl = "http://baseUrl";
        client.DefaultRequestHeaders.Add("key", "value");
        ...
    );
```

What are my options right now? What am I missing?

Pass around an object that holds my dynamic headers and then use ut like:
```
myHttpClient.ApiKey = MyHeadersObject.ApiKey;
myHttpClient.GetFoo();
```
?

**Comment by dannythunder** [comment, marked as answer]
Oh. I found out. Was not to clear to me.

```
services.AddRestEaseClient< IMyHttpClient >(baseUrl, new()
            {
                InstanceConfigurer = instance => instance.ApiKey = "myKey"
            });
```


```
public interface IMyHttpClient
{
    [Header("apikey")]
    string ApiKey { get; set; }
    
    [Get("/v1/one/ring/to/rule/them/all")]
    Task<ring> Get(CancellationToken cancellationToken);
}
```

**Comment by canton7** [comment]
You can configure the HttpClient as your question proposes (there's an overload of AddRestEaseClient which lets you do that), or you can add tha header as a property on your interface, and then there's an overload of AddRestEaseClient which lets you set that property when the interface is instantiated.

**Reply by canton7** [reply]
Looks like I didn't add InstanceConfigurer to the readme though, I'll do that

**Reply by canton7** [reply]
#242

**Reply by dannythunder** [reply]
I, just figured it out!

Thanks for this library! Just moved from Refit, pretty easy transition. Wish I found this before Refit (also a breeze, but lacks in udates)
